From ca116778846f3c50cbe41b00e6e72218b263de5e Mon Sep 17 00:00:00 2001
From: "darshak.patel" <darshak.patel@einfochips.com>
Date: Tue, 20 Oct 2020 15:48:36 +0530
Subject: [PATCH] IMX8 AIML: Enable EEPROM support

Command for testing :
For Write
$ echo arjun > /sys/bus/i2c/devices/16-0050/eeprom

For read
$ cat /sys/bus/i2c/devices/16-0050/eeprom | hexdump -C
---
 arch/arm64/boot/dts/freescale/imx8x-aiml-ei.dtsi | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/imx8x-aiml-ei.dtsi b/arch/arm64/boot/dts/freescale/imx8x-aiml-ei.dtsi
index 467dc648338a..ec5b0940ba17 100644
--- a/arch/arm64/boot/dts/freescale/imx8x-aiml-ei.dtsi
+++ b/arch/arm64/boot/dts/freescale/imx8x-aiml-ei.dtsi
@@ -152,6 +152,13 @@
	sda-gpios = <&lsio_gpio1 9 GPIO_ACTIVE_HIGH>;
	status = "okay";

+	eeprom: eeprom@50 {
+		compatible = "atmel,24c256";
+		reg = <0x50>;
+		pagesize = <32>;
+		status = "okay";
+	};
+
	ov5640: ov5640@3c {
		compatible = "ovti,ov5640";
		reg = <0x3c>;
--
2.17.1
